src/TuxxAnimations.js

```diff
@@ -49,59 +50,323 @@ var createAnimationClass = function (transitions, customClassName) {
+          }
+
+          //vector prop work--------------------------------------------
+          //vector property takes an object consisting of origin and magnitude
+
+          //origin prop work--------------------------------------------
+          var origin = this.props.origin
+          //check if origin is an accepted origin or throw error
+          var AcceptedOrigins = ['top','right','bottom','left','center','center center','top left','top center','top right','right center','bottom right','bottom center','bottom left', 'left center']
+          if (origin) {
+            if (AcceptedOrigins.indexOf(origin) > -1) {
+              enter['transform-origin'] = origin;
+              enterActive['transform-origin'] = origin;
+              leave['transform-origin'] = origin;
+              leaveActive['transform-origin'] = origin;
+            } else {
+              throw "origin type: animation origin property is not of acceptable type. List of acceptable origins: 'center center','top left','top center','top right','right center','bottom right','bottom center','bottom left', 'left center'";
+            }
+          }
+          if (enterProps) {
+            if (AcceptedOrigins.indexOf(enterProps.vector.origin) > -1) {
+              enter['transform-origin'] = enterProps.vector.origin;
+              enterActive['transform-origin'] = enterProps.vector.origin;
+            } else {
+              throw "origin type: animation origin property is not of acceptable type. List of acceptable origins: 'center center','top left','top center','top right','right center','bottom right','bottom center','bottom left', 'left center'";
+            }
+          }
+          if (leaveProps) {
+            if (AcceptedOrigins.indexOf(leaveProps.vector.origin) > -1) {
+              leave['transform-origin'] = leaveProps.vector.origin;
+              leaveActive['transform-origin'] = leaveProps.vector.origin;
+            } else {
+              throw "origin type: animation origin property is not of acceptable type. List of acceptable origins: 'center center','top left','top center','top right','right center','bottom right','bottom center','bottom left', 'left center'";
+            }
+          }
+
+          //magnitude prop work--------------------------------------------
+          //setMagnitude function
+          var setMagnitude = function (originType, origin, magnitude) {
+            //set transform prop in all transition states to empty string so as to concat translate animations
+            enter['transform'] = enter['transform'] || '';
+            enterActive['transform'] = enterActive['transform'] || '';
+            leave['transform'] = leave['transform'] || '';
+            leaveActive['transform'] = leaveActive['transform'] || '';
+
+            if (!origin) {
+              throw 'origin reference: the magnitude property requires an origin property to be set in order to function'
```

Comment:
Instead of throwing errors, use invariant. Look at other files in the tuxx/src for examples.